src/main/java/seedu/rc4hdb/ui/history/CommandHistoryParser.java

```diff
@@ -0,0 +1,61 @@
+package seedu.rc4hdb.ui.history;
+
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+import javafx.scene.input.KeyCode;
+
+/**
+ * The component responsible for logging valid commands, and parsing key press actions solely
+ * for viewing the command history.
+ */
+public class CommandHistoryParser {
+
+    private static Logger logger = Logger.getLogger("CommandHistoryParser");
+    private static CommandHistoryParser commandHistoryParser = null;
```

Comment:
I think you can just use a public static final field here for the reusable instance, and avoid setting the variable to null altogether.